src/ConfigurationClientManager.ts

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AppConfigurationClient, AppConfigurationClientOptions } from "@azure/app-configuration";
+import { ConfigurationClientWrapper } from "./ConfigurationClientWrapper.js";
+import { TokenCredential } from "@azure/identity";
+import { AzureAppConfigurationOptions, MaxRetries, MaxRetryDelayInMs } from "./AzureAppConfigurationOptions.js";
+import { isBrowser, isWebWorker } from "./requestTracing/utils.js";
+import * as RequestTracing from "./requestTracing/constants.js";
+
+const TCP_ORIGIN_KEY_NAME = "_origin._tcp";
+const ALT_KEY_NAME = "_alt";
+const TCP_KEY_NAME = "_tcp";
+const ENDPOINT_KEY_NAME = "Endpoint";
+const ID_KEY_NAME = "Id";
+const SECRET_KEY_NAME = "Secret";
+const TRUSTED_DOMAIN_LABELS = [".azconfig.", ".appconfig."];
+const FALLBACK_CLIENT_REFRESH_EXPIRE_INTERVAL = 60 * 60 * 1000; // 1 hour in milliseconds
+const MINIMAL_CLIENT_REFRESH_INTERVAL = 30 * 1000; // 30 seconds in milliseconds
+const SRV_QUERY_TIMEOUT = 30 * 1000; // 30 seconds in milliseconds
+
+export class ConfigurationClientManager {
+    isFailoverable: boolean;
+    dns: any;
+    endpoint: string;
+    #secret : string;
+    #id : string;
+    #credential: TokenCredential;
+    #clientOptions: AppConfigurationClientOptions | undefined;
+    #appConfigOptions: AzureAppConfigurationOptions | undefined;
+    #validDomain: string;
+    #staticClients: ConfigurationClientWrapper[];
+    #dynamicClients: ConfigurationClientWrapper[];
+    #lastFallbackClientRefreshTime: number = 0;
+    #lastFallbackClientRefreshAttempt: number = 0;
+
+    constructor (
+        connectionStringOrEndpoint?: string | URL,
+        credentialOrOptions?: TokenCredential | AzureAppConfigurationOptions,
+        appConfigOptions?: AzureAppConfigurationOptions
+    ) {
+        let staticClient: AppConfigurationClient;
+
+        if (typeof connectionStringOrEndpoint === "string" && !instanceOfTokenCredential(credentialOrOptions)) {
+            const connectionString = connectionStringOrEndpoint;
+            this.#appConfigOptions = credentialOrOptions as AzureAppConfigurationOptions;
+            this.#clientOptions = getClientOptions(this.#appConfigOptions);
+            staticClient = new AppConfigurationClient(connectionString, this.#clientOptions);
```

Comment:
This line should be placed after the error throwing of invalid connection string.